dateparser/date_parser.py

```diff
@@ -32,10 +33,82 @@ def __add__(self, other):
 parser.relativedelta.relativedelta = new_relativedelta
 
 
+class new_parser(parser.parser):
+    """
+    Implements an alternate parse method which supports preference to dates in future and past.
+    For more see issue #36
+    """
+
+    def parse(self, timestr, default=None, ignoretz=False, **kwargs):
+        # timestr needs to be a buffer as required by _parse
+        if isinstance(timestr, binary_type):
+            timestr = timestr.decode()
+
+        # Parse timestr
+        res = self._parse(timestr, **kwargs)
+
+        if res is None:
+            raise ValueError("unknown string format")
+
+        # Fill in missing date
+        new_date, given_fields = self.populate(res, default)
+
+        # Correct date to if prefer in future of past
+        new_date = self.correct(new_date, given_fields, default)
+
+        # Clean hour and minutes, etc in case not defined
+        if not res.hour and not res.minute and not res.second and not res.microsecond:
+            new_date = new_date.replace(hour=0, minute=0, second=0, microsecond=0)
+
+        return new_date
+
+    def populate(self, res, default):
+        given_fields = []  # fields that cannot be overriden
+        new_date = default
+
+        # Populate all fields
+        repl = {}
+        for field in ['year', 'month', 'day', 'hour', 'minute', 'second', 'microsecond']:
+            value = getattr(res, field)
+            if value is not None:
+                repl[field] = value
+                given_fields.append(field + 's')
```

Comment:
Instead of defining `given_fields` you can use keys of `repl` to iterate over and later `relativedelta(**{field + 's': 1})`